filer/admin/clipboardadmin.py

```diff
@@ -158,8 +166,19 @@ def ajax_upload(self, request, folder_id=None):
                     ', '.join(errors)) for field, errors in uploadform.errors.items()
                 ])
                 raise UploadException(self.messages['request-invalid'].format(form_errors))
-        except UploadException, e:
-            return HttpResponse(json.dumps({'error': unicode(e)}),
+        except UploadException, exception:
+            return HttpResponse(json.dumps({'error': unicode(exception)}),
+                                content_type=mimetype)
+        except Exception as error: # no matter the error, we don't return a 500 code
+            # an error occurred trying to build the file obj and the clipboard item
+            # since they are interconnected, we'll delete both to cleanup
+            if file_obj and file_obj.file:
+                file_obj.file.close()
+                file_obj.delete()
```

Comment:
can we move this in the block 
```python
if clipboard_item:
```
and leave the rest to be executed in the finally block